tests/test_fr_page.py

```diff
@@ -0,0 +1,74 @@
+# Tests for parsing a page
+#
+# Copyright (c) 2021 Tatu Ylonen.  See file LICENSE and https://ylonen.org
+
+import unittest
+
+from unittest.mock import patch
+from wikitextprocessor import Page, Wtp
+
+from wiktextract.config import WiktionaryConfig
+from wiktextract.extractor.fr.page import parse_page
+from wiktextract.thesaurus import close_thesaurus_db
+from wiktextract.wxr_context import WiktextractContext
+
+
+class FrPageTests(unittest.TestCase):
+    def setUp(self):
+        self.maxDiff = None
+        conf1 = WiktionaryConfig(
+            dump_file_lang_code="fr",
+            capture_language_codes=None,
+            capture_translations=True,
+            capture_pronunciation=True,
+            capture_linkages=True,
+            capture_compounds=True,
+            capture_redirects=True,
+            capture_examples=True,
+        )
+        self.wxr = WiktextractContext(Wtp(lang_code="fr"), conf1)
+
+    def tearDown(self) -> None:
+        self.wxr.wtp.close_db_conn()
+        close_thesaurus_db(
+            self.wxr.thesaurus_db_path, self.wxr.thesaurus_db_conn
+        )
+
+    def mock_clean_node(self, *args, **kwargs):
+        try:
+            if args[2].template_name == "langue":
+                return "Français"
+        except:
+            return "Nom commun"
```

Comment:
You could add these two templates using `Wtp.add_page()` and replace the page body with plain text(don't call lua code) before calling `parse_page()`.

Reply:
Yes, that's more elegant.